lib/trader.js

```diff
@@ -694,8 +698,50 @@ Trader.prototype.pollHandler = function pollHandler (data) {
     this.zeroConfLimits = _.mapValues(_.get('zeroConfLimit'), coins)
   }
 
+  /*
+   * There are 4 possibilities here:
+   *
+   * Nothing
+   *   The T&C don't exist or were removed/disabled.
+   *
+   * Just (Nothing, Nothing)
+   *   Shouldn't happen! Treated as if there were no T&C.
+   *
+   * Just (Just hash, Nothing)
+   *   `hash` should be equal to the cached one -- meaning the T&C haven't
+   *   changed.
+   *
+   * Just (Just hash, Just tc)
+   *   `hash` should differ from the cached one -- meaning the T&C have
+   *   changed.
+   */
+  const handleTerms = terms => {
+    if (!terms || !terms.hash) { // T&C don't exist or were removed/disabled
+      this.terms = false
+      this.termsHash = null
+      return
+    }
+
+    if (terms.hash === this.termsHash) return // T&C haven't changed
+
+    if (!terms.details) {
+      // No details even though there's a hash and it differs -- shouldn't happen!
+      this.terms = false
+      this.termsHash = null
+      return
+    }
+
+    this.terms = _.flow(
+      _.pick(['delay', 'title', 'text', 'accept', 'cancel']),
+      _.set('active', true)
```

Comment:
Why are we forcing active to true?

Reply:
The server send the details only if the T&C are active, but it doesn't send the active flag itself because it's implied.